shell/browser/zoom_level_delegate.cc

```diff
@@ -84,50 +76,48 @@ void ZoomLevelDelegate::OnZoomLevelChanged(
 
   double level = change.zoom_level;
   DictionaryPrefUpdate update(pref_service_, kPartitionPerHostZoomLevels);
-  base::Value* host_zoom_dictionaries = update.Get();
-  DCHECK(host_zoom_dictionaries);
+  base::Value* host_zoom_update = update.Get();
+  DCHECK(host_zoom_update);
+  base::Value::Dict& host_zoom_dictionaries = host_zoom_update->GetDict();
 
   bool modification_is_removal =
       blink::PageZoomValuesEqual(level, host_zoom_map_->GetDefaultZoomLevel());
 
-  base::Value* host_zoom_dictionary =
-      host_zoom_dictionaries->FindDictKey(partition_key_);
+  base::Value::Dict* host_zoom_dictionary =
+      host_zoom_dictionaries.FindDict(partition_key_);
   if (!host_zoom_dictionary) {
-    host_zoom_dictionaries->SetKey(partition_key_,
-                                   base::Value(base::Value::Type::DICTIONARY));
-    host_zoom_dictionary = host_zoom_dictionaries->FindDictKey(partition_key_);
+    base::Value::Dict dict;
+    host_zoom_dictionaries.Set(partition_key_, std::move(dict));
+    host_zoom_dictionary = host_zoom_dictionaries.FindDict(partition_key_);
   }
 
   if (modification_is_removal)
-    host_zoom_dictionary->RemoveKey(change.host);
+    host_zoom_dictionary->Remove(change.host);
   else
-    host_zoom_dictionary->SetKey(change.host, base::Value(level));
+    host_zoom_dictionary->Set(change.host, base::Value(level));
 }
 
 void ZoomLevelDelegate::ExtractPerHostZoomLevels(
-    const base::DictionaryValue* host_zoom_dictionary) {
+    const base::Value::Dict& host_zoom_dictionary) {
   std::vector<std::string> keys_to_remove;
-  std::unique_ptr<base::DictionaryValue> host_zoom_dictionary_copy =
-      host_zoom_dictionary->DeepCopyWithoutEmptyChildren();
-  for (base::DictionaryValue::Iterator i(*host_zoom_dictionary_copy);
-       !i.IsAtEnd(); i.Advance()) {
-    const std::string& host(i.key());
-    const absl::optional<double> zoom_level = i.value().GetIfDouble();
+  base::Value::Dict host_zoom_dictionary_copy = host_zoom_dictionary.Clone();
+  for (auto [host, value] : host_zoom_dictionary_copy) {
+    const absl::optional<double> zoom_level = value.GetIfDouble();
 
     // Filter out A) the empty host, B) zoom levels equal to the default; and
     // remember them, so that we can later erase them from Prefs.
     // Values of type B could further have been stored before the default zoom
     // level was set to its current value. In either case, SetZoomLevelForHost
     // will ignore type B values, thus, to have consistency with HostZoomMap's
     // internal state, these values must also be removed from Prefs.
-    if (host.empty() || !zoom_level ||
-        blink::PageZoomValuesEqual(*zoom_level,
+    if (host.empty() || !zoom_level.has_value() ||
+        blink::PageZoomValuesEqual(zoom_level.value(),
```

Comment:
fwiw i think the left side here works just as well as the right side! casting absl::optional to bool is the same as `.has_value()`, and `operator*` is the same as `.value()`.